// react-native-password-field/src/PasswordField.js
import React from 'react';
import {Button, Input, Icon} from 'native-base';

import {View} from 'react-native';

export default class PasswordField extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isSecureText: true,
      text: '',
    };
  }

  toggleSecureState() {
    this.setState({
      isSecureText: !this.state.isSecureText,
    });
  }

  render() {
    this.state.text = this.props.passwordValue;
    return (
      <View
        style={{
          backgroundColor: this.props.backgroundColor,
          height: 40,
          width: '100%',
          flexDirection: 'row',
          justifyContent: 'space-between',
          alignItems: 'center',
          alignSelf: 'stretch',
        }}>
        <Input
          placeholder={this.props.placeholder}
          autoCorrect={false}
          underlineColorAndroid="transparent"
          returnKeyType="go"
          secureTextEntry={this.state.isSecureText}
          value={this.state.text}
          onChangeText={value => {
            this.setState({text: value});
            this.props.onChangeText(value);
          }}
          style={{
            color: this.props.textColor,
            height: 20,
          }}
        />
        <Button
          transparent
          onPress={() => {
            this.toggleSecureState();
          }}
          style={{
            justifyContent: 'center',
            alignItems: 'center',
            height: 38,
          }}>
          <Icon
            name={this.state.isSecureText ? 'ios-eye' : 'ios-eye-off'}
            style={{
              color: 'blue',
              fontSize: 20,
              marginRight: 16,
              marginLeft: 16,
            }}
          />
        </Button>
      </View>
    );
  }
}
